### PR TITLE
Add pagination to the GET api/v2/projects/ endpoint

### PR DESCRIPTION
**Pull request by jeremycline**
This contains 3 commits, which should all be atomic and self-contained, so you can review them separately (oldest to newest, obviously).

It moves the scoped session creation to the model module, then adds a query property to models with a custom Query class that provides pagination for all models. Finally, it uses this in the new REST API.

### CODE REVIEW

File: anitya/api_v2.py [hunk 1]
```diff
@@ -18,19 +18,100 @@
 _BASE_ARG_PARSER.add_argument('access_token', type=str)
 
 
+def _page_validator(arg):
+    """Validator for a pagination page number."""
+    arg = int(arg)
```
Comment (pypingou) [inline anitya/api_v2.py]: What is int() raises an exception, do we want to handle that here?

Comment (jeremycline) [inline anitya/api_v2.py]: No, the flask-restful middleware will catch the exception and turn it into a pretty HTTP 400 with details about what went wrong. I did neglect to add tests for that though (which is why codecov was displeased with me). You can check out the tests to see what happens when the cast to int fails :)

File: anitya/api_v2.py [hunk 2]
```diff
@@ -18,19 +18,100 @@
 _BASE_ARG_PARSER.add_argument('access_token', type=str)
 
 
+def _page_validator(arg):
+    """Validator for a pagination page number."""
```
Comment (bowlofeggs) [inline anitya/api_v2.py]: It would be good to docblock the ```arg``` arg.

File: anitya/api_v2.py [hunk 3]
```diff
@@ -18,19 +18,100 @@
...
+    arg = int(arg)
+    if arg < 1:
+        raise ValueError(_('Value must be greater than 1'))
```
Comment (bowlofeggs) [inline anitya/api_v2.py]: Technically, greater than or equal to judging by the code.

File: anitya/api_v2.py [hunk 4]
```diff
@@ -18,19 +18,100 @@
...
+    return arg
+
+
+def _items_per_page_validator(arg):
+    """Validator for a pagination items_per_page number."""
```
Comment (bowlofeggs) [inline anitya/api_v2.py]: It would be good to docblock the ```arg``` arg.

File: anitya/api_v2.py [hunk 5]
```diff
@@ -18,19 +18,100 @@
...
+    arg = int(arg)
+    if arg < 1:
+        raise ValueError(_('Value must be greater than 1'))
```
Comment (bowlofeggs) [inline anitya/api_v2.py]: greater than or equal to

File: anitya/api_v2.py [hunk 6]
```diff
@@ -18,19 +18,100 @@
...
+    if arg > 250:
+        raise ValueError(_('Value must be less than 250'))
```
Comment (bowlofeggs) [inline anitya/api_v2.py]: Less than or equal to.

File: anitya/api_v2.py [hunk 7]
```diff
@@ -18,19 +18,100 @@
...
+            }
+
+
+        :query int page: The project page number to retrieve (defaults to 1).
+        :query int items_per_page: The number of items per page (defaults to
+                                   25, maximum of 250).
+        :statuscode 200: If all arguments are valid. Note that even if there
+                         are no projects, this will return 200.
+        :statuscode 400: If one or more of the query arguments is invalid.
+        """
```
Comment (bowlofeggs) [inline anitya/api_v2.py]: Very nice docblock!

File: anitya/config.py [hunk 8]
```diff
@@ -21,6 +21,7 @@
 
 from datetime import timedelta
 import logging
+import logging.config
```
Comment (bowlofeggs) [inline anitya/config.py]: This change seems unrelated to the commit. Is it needed? I recommend making a separate commit with it if it is.

File: anitya/lib/model.py [hunk 9]
```diff
@@ -70,7 +71,88 @@ def initialize(config):
...
+    A sub-class of namedtuple that represents a page.
+
+    Attributes:
+        items (object): The database objects from the query.
+        page (int): The page number used for the query.
+        items_per_page (int): The number of items per page.
+        total_items (int): The total number of items in the database.
+    """
+
+    def as_dict(self):
```
Comment (bowlofeggs) [inline anitya/lib/model.py]: It would be good to docblock this, even though it's fairly obvious.
